R/render.R

```diff
@@ -23,6 +23,10 @@
 render_page <- function(pkg = ".", name, data, path = "", depth = NULL, quiet = FALSE) {
   pkg <- as_pkgdown(pkg)
 
+  if (is.null(data$has_toc)) {
+    data$has_toc <- TRUE
```

Comment:
One thing I only tested quickly is creating a vignette with tabsets and no h2: then the tabs work because there's a TOC even if it's empty.